roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -0,0 +1,170 @@
+    is_writable = False
+    mode = 'file'
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, filepath):
+        """
+        Parameters
+        ----------
+        filepath: str
+            The location of the folder containing caiman *.hdmf output file.
+        """
+        SegmentationExtractor.__init__(self)
+        self.filepath = filepath
+        self._dataset_file = self._file_extractor_read()
+        self._roi_response = self._trace_extractor_read('F_dff')
+        self._roi_response_fluorescence = self._roi_response
+        self._roi_response_neuropil = self._trace_extractor_read('C')
+        self._roi_response_deconvolved = self._trace_extractor_read('S')
+        self._images_correlation = self._summary_image_read()
+        self._raw_movie_file_location = self._dataset_file['params']['data']['fnames'][0].decode('utf-8')
+        self._sampling_frequency = self._dataset_file['params']['data']['fr'][()]
+        self.image_masks = None
+
+    def __del__(self):
+        self._dataset_file.close()
+
+    def _file_extractor_read(self):
+        f = h5py.File(self.filepath, 'r')
+        return f
+
+    def _image_mask_sparse_read(self):
+        roi_ids = self._dataset_file['estimates']['A']['indices']
+        masks = self._dataset_file['estimates']['A']['data']
+        ids = self._dataset_file['estimates']['A']['indptr']
+        return masks, roi_ids, ids
+
+    def _trace_extractor_read(self, field):
+        extracted_signals = self._dataset_file['estimates'][field] # lazy read dataset)
+        return extracted_signals
+
+    def _summary_image_read(self):
+        if self._dataset_file['estimates'].get('Cn'):
+            summary_images_ = self._dataset_file['estimates']['Cn']
+            return np.array(summary_images_).T
+        else:
+            return None
+
+    def get_accepted_list(self):
+        accepted = self._dataset_file['estimates']['idx_components']
+        if len(accepted.shape)==0:
+            accepted = list(range(self.no_rois))
+        return accepted
+
+    def get_rejected_list(self):
+        return [a for a in range(self.no_rois) if a not in set(self.get_accepted_list())]
+
+    @property
+    def roi_locations(self):
+        _masks, _mask_roi_ids, _mask_ids = self._image_mask_sparse_read()
+        roi_location = np.ndarray([2, self.no_rois], dtype='int')
+        for i in range(self.no_rois):
+            max_mask_roi_id = _mask_roi_ids[_mask_ids[i]+np.argmax(
+                _masks[_mask_ids[i]:_mask_ids[i+1]]
+            )]
+            roi_location[:, i] = [((max_mask_roi_id+1)%(self.image_size[0]+1))-1,#assuming order='F'
+                                  ((max_mask_roi_id+1)//(self.image_size[0]+1))]
+            if roi_location[0,i]<0:
+                roi_location[0,i]=0
+        return roi_location
+
+    @staticmethod
+    def write_segmentation(segmentation_object, savepath, **kwargs):
+        from scipy.sparse import csc_matrix
```

Comment:
strange placement for this import. Is there a reason this is here and not at the top?

Reply:
yeah, i had initially kept it since scipy is used nowhere else in the repo. But since its a common package, it makes sense to add that as a dependency overall and move it on top